### PR TITLE
Fix unicode errors & refactor script router

### PR DESCRIPTION
**Pull request by dplim87**
Fixed some unicode errors, reduced a couple function calls and refactored script router to use routing table (should be faster)

### CODE REVIEW

**Review by jurialmunkey — CHANGES_REQUESTED** [review]
Couple of comments, but I like this approach overall! Probably should do something similar in the container router too.

File: resources/lib/script/router.py [hunk 1]
```diff
@@ -267,73 +267,53 @@ def sort_list(**kwargs):
...
+    def __init__(self):
+        self.params = {}
+        for arg in [try_decode(arg) for arg in sys.argv[1:]]:
+            if '=' in arg:
+                key, value = arg.split('=', 1)
+                self.params[key] = value.strip('\'').strip('"') if value else True
             else:
-                params.setdefault(arg, True)
-        return params
+                self.params[key] = True
```
[the remark this answers is not in the document]
Comment (dplim87) [reply resources/lib/script/router.py:277]: oops, yes

File: resources/lib/script/router.py [hunk 2]
```diff
@@ -267,73 +267,53 @@ def sort_list(**kwargs):
...
-            return log_request(**self.params)
-        if self.params.get('play'):
-            return play_external(**self.params)
         if self.params.get('restart_service'):
             # Only do the import here because this function only for debugging purposes
             from resources.lib.monitor.service import restart_service_monitor
             return restart_service_monitor()
+
+        routes_available = set(self.routing_table.viewkeys())
+        params_given = set(self.params.viewkeys())
```
[the remark this answers is not in the document]
Comment (dplim87) [reply resources/lib/script/router.py:316]: actually, just need to change from dict.viewkeys() to dict.keys(). py2 has dict.keys()
oops again

**Comment by drinfernoo** [comment]
Ooh, yeah, that's nice!

**Review by dplim87 — COMMENTED** [review]
should fix things? work on container tomorrow

**Comment by jurialmunkey** [comment]
Merged

**Pull request closed without merging**